editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -249,14 +246,35 @@ export const previewStore: StateCreator<
           {} as Store.passport["data"],
         );
 
-        return {
+        const passport: Store.passport = {
           ...acc,
           data: {
             ...acc.data,
             ...responseData,
             ...passportData,
           },
         };
+
+        const isSetValue = flow[id].type === TYPES.SetValue;
+
+        if (isSetValue) {
+          const { operation, fn }: { operation: string; fn: string } =
+            flow[id]?.data || {};
+
+          if (operation === "remove") {
+            delete passport.data?.[fn];
+          }
+
+          if (operation === "append") {
+            const previousValue = acc.data?.[fn] || [];
+            const currentValue = responseData?.[fn] || [];
+            const combined = [...previousValue, ...currentValue];
+
+            passport.data![fn] = combined;
+          }
+        }
```

Comment:
I'm not massively keen on making `computePassport()` much longer or more complex, but hopefully this is fairly self contained?

If inclear, happy to move out to a helper function!

Reply:
Rather than adding this logic to the Preview store, what if it lived directly in SetValue's `handleSubmit` ? Eg similar to Calculate and the use of `props.formatOutputForAutomations`? 

Interested in a future of smarter components, dumber stores!

Reply:
Caught up about this one to discuss options. This isn't easily possible as it would require modifying past breadcrumbs to effectively handle the "remove" operation.